lib/report-generator.js

```diff
@@ -0,0 +1,95 @@
+var fs = require('fs-extra'),
+    path = require('path');
+
+/**
+ * The directory of the visual report
+ *
+ * @type {String}
+ * @const
+ * @default
+ */
+var DIRECTORY_PATH = 'visual-report';
+
+/**
+ * The directory of the static files
+ *
+ * @type {String}
+ * @const
+ * @default
+ */
+var STATIC_DIRECTORY = 'statics';
+
+/**
+ * The path of the data file
+ *
+ * @type {String}
+ * @const
+ * @default
+ */
+var DATA_PATH = 'data.js';
+
+/**
+ * The root directory
+ *
+ * @type {String}
+ */
+var ROOT_DIRECTORY;
+
+
+/**
+ * Creates the files for the visual report
+ *
+ * @param {Array.<Suite>} data - The generated data from Mocha's runner
+ * @param {Object} options - Options object
+ * @param {String} options.rootDirectory - The directory of the visual report
+ * @param {generateReportCb} done - Cb to notify when the report is created
+ */
+module.exports = function(data, options, done) {
+  if (data === undefined) {
+    throw new Error('No received data to create the "data.js" file');
+  }
+
+  if (typeof options === 'function') {
```

Comment:
Yes, we should remove it because we don't have a clear understanding of how this module would be used in this case.